### PR TITLE
Document Pipeline

### PR DESCRIPTION
**Pull request by eyurtsev**
Create a simple pipeline to help compose loaders and transformers to yield document loaders.

### CODE REVIEW

File: libs/langchain/langchain/document_loaders/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            documents = self.loader.lazy_load()
+        except NotImplementedError:
+            documents = iter(self.loader.load())
+
+        for document in documents:
+            _docs = [document]
+            for transformer in self.transformers:
+                # List below is needed because of typing issue in langchain
+                _docs = list(transformer.transform_documents(_docs))
+            yield from _docs
```
Comment (leo-gan) [inline libs/langchain/langchain/document_loaders/pipeline.py:43]: Maybe, 

> for document in documents:
>      applying all transformers to a single doc then yield it

Now, it is not really lazy.

**Comment by leo-gan** [comment]
Can we add also text splitters to this pipeline? I'm not sure this addition fits the current design, just a thought.

File: libs/langchain/langchain/document_loaders/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    def load(self) -> List[Document]:
+        """Fetch the data from the data selector."""
+        raise NotImplementedError("Use lazy_load instead")
```
Comment (baskaryan) [inline libs/langchain/langchain/document_loaders/pipeline.py:47]: why no implement list(lazy_load)

Comment (eyurtsev) [reply libs/langchain/langchain/document_loaders/pipeline.py:47]: i'd prefer only generators to be supported to discourage anything that's non lazy

**Comment by eyurtsev** [comment]
Converted back into draft mode. We'll try to find a more general approach using runnables. Otherwise we're re-inventing sync/async/batch/abatch and concurrency every time. 

@leo-gan text splitters are instances of transformers so they will be accommodated by this design, but we're going to hold off for a bit to see if we can handle more known requirements for a pipeline design

**Review by baskaryan — COMMENTED** [review]
yea idk if this passes the $10 test 🙃

**Comment by leo-gan** [comment]
@eyurtsev If this PR is not needed anymore, could you, please, let me know or close this PR?

**Pull request closed without merging**